apps/dashboard/app/(app)/semantic-cache/analytics/page.tsx

```diff
@@ -15,70 +28,152 @@ const tokenCostMap = {
+    }
+    case "30d": {
+      now.setUTCDate(now.getUTCDate() + 1);
+      const end = now.setUTCHours(0, 0, 0, 0);
+      const intervalMs = 1000 * 60 * 60 * 24 * 30;
+      return {
+        start: end - intervalMs,
+        end,
+        intervalMs,
+        granularity: 1000 * 60 * 60 * 24,
+        getSemanticCachesPerInterval: getSemanticCachesDaily,
+      };
+    }
+    case "90d": {
+      now.setUTCDate(now.getUTCDate() + 1);
+      const end = now.setUTCHours(0, 0, 0, 0);
+      const intervalMs = 1000 * 60 * 60 * 24 * 90;
+      return {
+        start: end - intervalMs,
+        end,
+        intervalMs,
+        granularity: 1000 * 60 * 60 * 24,
+        getSemanticCachesPerInterval: getSemanticCachesDaily,
+      };
+    }
+  }
+}
+
+export default async function SemanticCacheAnalyticsPage(props: {
+  searchParams: { interval?: Interval };
+}) {
+  const interval = props.searchParams.interval ?? "24h";
+
+  const tenantId = getTenantId();
+  const workspace = await db.query.workspaces.findFirst({
+    where: (table, { and, eq, isNull }) =>
+      and(eq(table.tenantId, tenantId), isNull(table.deletedAt)),
+    with: {
+      llmGateways: {
+        columns: {
+          id: true,
+          name: true,
+        },
+      },
+    },
   });
 
-  console.info({ data });
+  if (!workspace) {
+    return redirect("/new");
+  }
+
+  const gatewayId = workspace?.llmGateways[0]?.id;
 
-  const tokenCost = data.reduce((acc, log) => acc + tokenCostMap[log.model].cost * log.tokens, 0);
-  const tokens = data.reduce((acc, log) => acc + log.tokens, 0);
-  const timeSaved = data.reduce((acc, log) => acc + log.tokens / tokenCostMap[log.model].tps, 0);
+  if (!gatewayId) {
+    return redirect("/semantic-cache/new");
+  }
+
+  const { start, end, getSemanticCachesPerInterval } = prepareInterval(interval);
 
-  console.info({ tokenCost, tokens, timeSaved });
+  const query = {
+    start,
+    end,
+    // TODO unhardcode
+    gatewayId,
+    workspaceId: workspace.id,
+  };
 
-  const transformedData = data.map((log) => {
-    const isCacheHit = log.cache > 0;
-    return {
-      x: log.time,
-      y: isCacheHit ? 1 : 0,
-      category: isCacheHit ? "cache hit" : "cache miss",
-    };
+  const { data: tokensData } = await getAllSemanticCacheLogs({
+    limit: 1000,
+    gatewayId,
+    workspaceId: workspace.id,
   });
 
-  const dailyCounts: { [key: string]: { [key: string]: number } } = {};
+  // const analyticsData = await _getSemanticCachesDaily();
+  const { data: analyticsData } = await getSemanticCachesPerInterval(query);
 
-  transformedData.forEach((entry) => {
-    const date = new Date(entry.x).toISOString().split("T")[0];
-    const category = entry.category;
-    if (!dailyCounts[date]) {
-      dailyCounts[date] = {};
-    }
-    if (!dailyCounts[date][category]) {
-      dailyCounts[date][category] = 0;
-    }
-    dailyCounts[date][category] += 1;
-  });
+  const tokens = tokensData.reduce((acc, log) => acc + log.tokens, 0);
```

Comment:
The new pipes have an `avgTokens` column, but now that I see this, I think maybe a `sumTokens` would be more useful 

wdyt?

Reply:
That would be helpful, we should be able to show all tokens cached per gatewayId